web_poet/pages.py

```diff
@@ -64,9 +71,80 @@ def __init_subclass__(cls, skip_nonitem_fields: bool = False, **kwargs):
     async def to_item(self) -> ItemT:
         """Extract an item from a web page"""
         return await item_from_fields(
-            self, item_cls=self.item_cls, skip_nonitem_fields=self._skip_nonitem_fields
+            self,
+            item_cls=self.item_cls,
+            skip_nonitem_fields=self._skip_nonitem_fields,
+            field_names=self.fields_to_extract,
         )
 
+    def _get_select_fields(self) -> Optional[SelectFields]:
+        select_fields = getattr(self, "select_fields", None)
+        if not select_fields:
+            return None
+
+        if not isinstance(select_fields, SelectFields):
+            return None
+
+        return select_fields
+
+    @property
+    def fields_to_extract(self) -> Iterable[str]:
+        """Returns an Iterable of field names which should populate the designated
+        ``item_cls``.
+
+        If :class:`web_poet.fields.SelectFields` is set, this takes into account
+        the fields that are marked as included or excluded alongside any field
+        that are marked as enabled/disabled by default.
+        """
+        select_fields = self._get_select_fields()
+        if select_fields is None:
+            return list(get_fields_dict(self))
+
+        fields = select_fields.fields
+
+        if fields is None or len(fields) == 0:
+            return list(get_fields_dict(self))
+
+        page_obj_fields = get_fields_dict(self, include_disabled=True)
+
+        unknown_fields = set(fields) - set(page_obj_fields.keys()).union({"*"})
+        if unknown_fields:
+            self._handle_unknown_field(unknown_fields, select_fields.on_unknown_field)
+
+        fields_to_extract = []
+        for name, field_info in page_obj_fields.items():
+            if fields.get("*") is False and fields.get(name) is not True:
+                continue
+            if (
+                fields.get("*") is True
+                or fields.get(name) is True
+                or field_info.disabled is False
+            ) and fields.get(name) is not False:
+                fields_to_extract.append(name)
+
+        return fields_to_extract
+
+    def _handle_unknown_field(
+        self, name: Set[str], action: UnknownFieldActions = "raise"
+    ) -> None:
+        if action == "ignore":
+            return None
+
+        msg = (
+            f"The {name} fields isn't available in {get_fq_class_name(self.__class__)}."
+        )
+
+        if action == "raise":
+            raise AttributeError(msg)
+        elif action == "warn":
+            warn(msg)
+        else:
+            raise ValueError(
```

Comment:
I'm thinking if we still need this runtime check somehow since mypy should be good enough to detect invalid values due to the `Literal` typing.

Reply:
Removed it.